### ISSUE TITLE
Preserve focus in DetailsList when items change

### ISSUE THREAD

**Issue by elliottsj** [OfficeDev/office-ui-fabric-react#3325]
To satisfy accessibility spec MAS 2.4.3, DetailsList focus should be restored to the row at the previously-selected index when the focused row is deleted.
